utils.sh

```diff
@@ -21,3 +21,65 @@ for patch in "${PWD}/ignition_patches/${kind}/"*.json; do
 done
 sudo cp "$current" "$target"
 }
+
+function dom_mac_retrieve() {
+    local domain
+    local domainxml
+    local rc
+
+    domain="$1"
+    domainxml="$(sudo virsh dumpxml "$domain")"
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        return $rc
+    fi
+    echo "$domainxml" | python <(cat << 'EOF'
+import sys
+from lxml import etree
+
+bridge = sys.argv[2]
+domain = etree.parse(sys.stdin)
+mac = domain.xpath("//devices/interface[source/@bridge='%s']/mac/@address" % bridge)
+if len(mac) < 1:
+    sys.stderr.write('No interface found for bridge %s\n' % bridge)
+    sys.exit(1)
+elif len(mac) > 1:
+    sys.stderr.write('Multiple interfaces per network not supported\n')
+    sys.exit(1)
+else:
+    print(mac[0])
+EOF
+    ) "$@"
+
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        return $rc
+    fi
+}
+
+function net_iface_dhcp_ip() {
+local netname
+local hwaddr
+
+netname="$1"
+hwaddr="$2"
+sudo jq -r --arg HWADDR "$hwaddr" '.[] | select(."mac-address"==$HWADDR) | ."ip-address"' "/var/lib/libvirt/dnsmasq/${netname}.status"
```

Comment:
Could we use `virsh net-dhcp-leases <network>` instead of parsing the status file directly?